nautobot/extras/views.py

```diff
@@ -585,6 +638,27 @@ def get_extra_context(self, request, instance):
             else:
                 context["choices"] = forms.CustomFieldChoiceFormSet(instance=instance)
 
+        if self.action == "retrieve" and hasattr(instance, "custom_field_choices"):
+            context.update(
+                {
+                    "columns": [
+                        "value",
+                        "weight",
+                    ],
+                    "header": [
+                        "Value",
+                        "Weight",
+                    ],
+                    "choices_data": [
+                        {
+                            "value": choice.value,
+                            "weight": choice.weight,
+                        }
+                        for choice in instance.custom_field_choices.all()
+                    ],
+                }
+            )
```

Comment:
can we go with something a bit easier to read?, I think this is equivalant, but would need to test it.

```suggestion
             choices_data = []

             for choice in instance.custom_field_choices.all():
                 choices_data.append({"value": choice.value,"weight": choice.weight})

             context["columns"] = ["value", "weight"]
             context["header"] = ["Value", "Weight"]
             context["choices_data"] = choices_data
```